### ISSUE TITLE
Interpolation on two different meshes doesn't work on parallel

### ISSUE THREAD

**Issue by argyriospetras** [firedrakeproject/firedrake#3703]
I am trying to perform interpolation on two different meshes, and while everything works as expected in serial, I am getting an error when trying to run it in parallel. Here is a small example that reproduces the error on my computer. https://drive.google.com/drive/folders/100ThT1R7Fe-dKx_u4U5XPTdfuW6-Nxez?usp=sharing
I am getting an runtime error
`RuntimeError: Some points have been claimed by a cell that we cannot see, but which we think we own. This should not happen.
`
which appears to be coming from here https://www.firedrakeproject.org/_modules/firedrake/mesh.html.

**Comment by nabw** [comment]
Hi, +1 on this error as it would be great to have this feature. Maybe there are some existing workarounds?

**Comment by connorjward** [comment]
Closing as a duplicate of #3151. I am currently trying to fix this.
